docs/data/sdks/android-kotlin/migration.md

```diff
@@ -1,8 +1,242 @@
+description: Use this guide to easily migrate from Amplitude's maintenance Android SDK (com.amplitude:android-sdk) to the latest SDK (com.amplitude:analytics-android).
 ---
 
+Amplitude's latest Android SDK (`com.amplitude:analytics-android`) features a plugin architecture, built-in type definitions, is written in Kotlin. The latest Android SDK isn't backwards compatible with the maintenance Android SDK `com.amplitude:android-sdk`. 
+
+To migrate to `com.amplitude:analytics-android`, update your dependencies and instrumentation.
+
+## Terminology
+
+* `com.amplitude:android-sdk`: Maintenance Android SDK
+* `com.amplitude:analytics-android`: Latest Android SDK
+
+## Dependency
+
+Update build.gradle to remove the maintennace Android SDK and add the latest Android SDK. Then sync project with Gradle files.
+
+```diff
+dependencies {
+-  implementation 'com.amplitude:android-sdk:2.+'
+-  implementation 'com.squareup.okhttp3:okhttp:4.2.2'
++  implementation 'com.amplitude:analytics-android:1.+'
+}
+```
+
+## Instrumentation
+
+The latest Android SDK offers an new API to instrument events. To migrate to it, you need to update a few calls. The following sections detail which calls have changed.
+
+### Initialization
+
+You must initialize the SDK with a valid Amplitude API Key and Android application context.
+
+=== "Kotlin"
+
+    ```diff
+    - import com.amplitude.api.Amplitude
+    - import com.amplitude.api.AmplitudeClient
+    + import com.amplitude.android.Amplitude
+
+    - val client = Amplitude.getInstance()
+    -   .initialize(getApplicationContext(), "YOUR_API_KEY")
+    + val client = Amplitude(
+    +     Configuration(
+    +         apiKey = "YOUR_API_KEY",
+    +         context = getApplicationContext()
+    +     )
+    + )
+    ```
+
+=== "Java"
+
+    ```diff
+    - import com.amplitude.api.Amplitude;
+    - import com.amplitude.api.AmplitudeClient;
+    + import com.amplitude.android.Amplitude;
+
+    - AmplitudeClient client = Amplitude.getInstance()
+    -   .initialize(getApplicationContext(), "YOUR_API_KEY");
+    + Amplitude client =  new Amplitude(new Configuration(
+    +     apiKey = "YOUR_API_KEY",
+    +     context = getApplicationContext()
+    + ));
+    ```
+
+### Configuration
+
+The latest Android SDK configuration comes in a different shape. Some configurations are no longer supported.
+
+|com.amplitude:android-sdk|com.amplitude:analytics-android|
+|-|-|
+| `eventUploadPeriodMillis` | `flushIntervalMillis` |
+| `eventUploadThreshold` | `flushQueueSize` |
+| `eventUploadMaxBatchSize` | Not supported |
+| `eventMaxCount` | Not supported |
+| `identifyBatchIntervalMillis` | `identifyBatchIntervalMillis` |
+| `flushEventsOnClose` | `flushEventsOnClose` |
+| `optOut` | `optOut` |
+| `trackingSessionEvents` | `trackingSessionEvents` |
+| `sessionTimeoutMillis` | Not supported |
+| `minTimeBetweenSessionsMillis` | `minTimeBetweenSessionsMillis` |
```

Comment:
Still working on these two configs and trying to explain them clearly.

- legacy Android SDK has two configs based on whether foreground session tracking is enabled or not
![image](https://github.com/Amplitude-Developer-Docs/amplitude-dev-center/assets/31029607/e93d5dea-ca73-490c-ad9f-ff6089dc2a9e)

- latest Android SDK only and by default supports foreground session tracking with config `minTimeBetweenSessionsMillis`

Reply:
The description of these two configs of legacy Android should be swapped and fixed it in another PR https://github.com/Amplitude-Developer-Docs/amplitude-dev-center/pull/848